### ISSUE TITLE
Changes are not reflected locally 

### ISSUE THREAD

**Issue by timalsinab** [RunestoneInteractive/rs#429]
**What is your username**
@timalsinab @solijonovam 

**Describe the bug**
When working on a front-end issue, making changes locally is not reflected. In order for changes to be reflected, we need to rebuild and reload the containers again and again even for small changes. Documentation could be enhanced with clear instructions on how to test changes locally.

**Comment by bnmnetp** [comment]
Please see https://runestone-monorepo.readthedocs.io/en/latest/server_feature.html

If you think more needs to be said let me know.  If that clarifies things then we can close.

**Comment by bnmnetp** [comment]
You do not need to run `pip install rich`, if you activate the virtual environment with `poetry shell` from the rs folder.

Steps 4 and 5 are redundant.  The `--restart` flag tells the build script do do just that.

**Comment by timalsinab** [comment]
to clarify build instructions, would you like us to add step 2; we can also mention that you don't need to run pip install rich if you have a virtual environment activated. We can add these steps in the https://runestone-monorepo.readthedocs.io/en/latest/building_servers.html#building-and-running-the-servers.  These steps were recommended to us by a classmate who was successfully able to build after struggling to do so for a while.

**Comment by bnmnetp** [comment]
No, it stands for the service you are rebuilding and restarting.

**Comment by bnmnetp** [comment]
Most command line commands give you helpful information if you run with the `--help` flag.
```
./build.py --help                                                                                            ─╯
Checking your environment
Usage: build.py [--verbose] [--help] [--all] [--push]
        --all build all containers, including author and worker
        --push push all containers to a container registry
        --one <service> build just one container, e.g. --one author
        --restart restart the container(s) after building

        If something in the build does not work or you have questions about setup or environment
        variables or installation, please check out our developer documentation.
        https://runestone-monorepo.readthedocs.io/en/latest/developing.html
```

**Comment by timalsinab** [comment]
Thank you for replying. We will add the clarification to the https://runestone-monorepo.readthedocs.io/en/latest/server_feature.html.  Really appreciate your help.

**Comment by bnmnetp** [comment]
Closing as no longer applicable. The build workflow discussed in this thread doesn't exist any more: the project moved from poetry to uv, so `poetry shell` and `./build.py --one <service> --restart` are now `uv sync` and `uv run build --service <service> full`, and the documentation was rewritten to match.

Thanks for pushing on this — "how do I see my change" is exactly the right question for a new contributor to ask, and the discussion here did improve the docs at the time. If the current uv-based instructions still leave a gap, a new issue, or a PR against the docs, would be very welcome.
